packages/hash/shared/src/blockMeta.ts

```diff
@@ -47,52 +47,95 @@ export const componentIdToUrl = (componentId: string) =>
   componentId.replace(/\/$/, "");
 
 /**
- * transform mere options into a useable block configuration
+ * Get an absolute url if the path is not already one.
  */
-const toBlockConfig = (
-  options: BlockMetadata,
-  componentId: string,
-): BlockConfig => {
+function getAbsoluteUrl(args: { baseUrl: string; path: string }): string;
+function getAbsoluteUrl(args: {
+  baseUrl: string;
+  path?: string | null | undefined;
+}): string | null | undefined;
+function getAbsoluteUrl({
+  baseUrl,
+  path,
+}: {
+  baseUrl: string;
+  path?: string | null;
+}): string | null | undefined {
+  const regex = /^(?:[a-z]+:)?\/\//i;
+  if (!path || regex.test(path)) {
+    return path;
+  }
+
+  return `${baseUrl}/${path.replace(/^\//, "")}`;
+}
+
+/**
+ * Transform a block metadata and schema file into fully-defined block variants.
+ * This would ideally be the one place we manipulate block metadata.
+ */
+const toBlockConfig = ({
+  componentId,
+  metadata,
+  schema,
+}: {
+  componentId: string;
+  metadata: BlockMetadata;
+  schema: BlockMeta["componentSchema"];
+}): BlockConfig => {
+  const defaultProperties =
```

Comment:
This is probably out of scope, but I wonder if we can replace [block-metadata] → `default` with something more concrete like [block-metadata] → `defaultProperties` for example. Otherwise, it’s not 100% clear what is meant by `default`.

Reply:
Looking at the docs, `default` and `examples` seem to be property-level metadata fields rather than entity-level.

Unless I am missing something, it might be more intuitive to find `default(Properties)`, `variants` and `examples` in one place, i.e. in `block-metadata.json`. To be honest, I don't have a strong opinion on this 🙂